### PR TITLE
KOGITO-1901: Manage columns enhancement

### PR DESCRIPTION
**Pull request by Sara4994**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by AjayJagan — APPROVED** [review]
works well :+1:

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx [hunk 1]
```diff
@@ -330,7 +331,7 @@ const DomainExplorerTable = ({
                   No columns selected
                 </Title>
                 <EmptyStateBody>
-                  Select columns from the dropdown to see content
+                  Select columns from Manage Columns to see content
```
Comment (srambach) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx]: I think we can simplify the text, and we should make "Manage columns" a link. "[Manage columns](url) to see content"

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx [hunk 2]
```diff
@@ -300,40 +189,107 @@ const DomainExplorerManageColumns: React.FC<IOwnProps> = ({
...
+                  <DataListContent
+                    aria-label="Primary Content Details"
+                    id="ex-expand1"
+                    isHidden={!expanded.includes(label.replace(/\,/g, ''))}
+                    className="kogito-common--manage-columns__data-list-content"
+                    key={Math.random()}
+                  >
+                    <DataListItemRow>
+                      <DataListCheck
+                        aria-labelledby="table-column-management-item2"
```
Comment (srambach) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: This should match the id of the label for the checkbox. This is important for screen readers to know what to read out for the checkbox.

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx [hunk 3]
```diff
@@ -343,31 +299,59 @@ const DomainExplorerManageColumns: React.FC<IOwnProps> = ({
...
-                      {item.name}
-                    </SelectOption>
-                  ))}
-            </SelectGroup>
+                  .map((item, _index) => {
+                    allSelections.push(item.name + '/' + group.name);
+                    return (
+                      <DataListContent
+                        aria-label="Primary Content Details"
+                        id="ex-expand1"
```
Comment (srambach) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: ids need to be unique.

**Comment by srambach** [comment]
There are some a11y problems shown when checked with the Axe chrome extension. Most are related to ids that are not unique, and attributes that don't match a valid id. I think the easiest thing would be to install the Axe chrome extension in order to help find these. https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx (hunk 2 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: done

**Comment by Sara4994** [comment]
> There are some a11y problems shown when checked with the Axe chrome extension. Most are related to ids that are not unique, and attributes that don't match a valid id. I think the easiest thing would be to install the Axe chrome extension in order to help find these. https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US

fixed accessibility related errors on Manage columns and DE's empty state

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx [hunk 4]
```diff
@@ -166,15 +183,221 @@ describe('Domain Explorer manage columns component', () => {
...
+              'state',
+              'start',
+              'lastUpdate',
+              'businessKey'
+            ]
+          }
+        ]
+      }
+    };
+    // const event = {
```
Comment (jstastny-cz) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx]: remove commented code.

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx [hunk 5]
```diff
@@ -166,15 +183,221 @@ describe('Domain Explorer manage columns component', () => {
...
+      .find('DataListToggle')
+      .first()
+      .props()
+      .onClick(event2);
+    wrapper
+      .find('DataListToggle')
+      .at(2)
+      .props()
+      .onClick(event2);
+    // wrapper.find('DataListCheck').first().props().onChange(event);
```
Comment (jstastny-cz) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx]: remove commented code

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx [hunk 6]
```diff
@@ -316,23 +553,31 @@ describe('Domain Explorer manage columns component', () => {
...
-    await Promise.resolve();
+    const wrapper = mount(<DomainExplorerManageColumns {...props} />);
+    wrapper
+      .find('#manage-columns-button')
+      .first()
+      .simulate('click');
+    wrapper
+      .find('#save-columns')
+      .first()
+      .simulate('click');
```
Comment (jstastny-cz) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx:584]: do we want to check what happened here? something appeared/hid or is this testing that #save-columns appeared after first click and can be clicked? If so, I'd check that the button hid again.

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx (hunk 1 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx]: this is done

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx [hunk 7]
```diff
@@ -716,47 +1020,85 @@ describe('Domain Explorer manage columns component', () => {
...
+    wrapper
+      .find('#manage-columns-button')
+      .first()
+      .simulate('click');
+    expect(wrapper.find('#selectAll-dropdown')).toBeTruthy();
+    wrapper
+      .find('#selectAll-dropdown')
+      .first()
+      .props()
+      ['toggle']['props']['splitButtonItems'][0]['props']['onClick']();
```
Comment (jstastny-cz) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx:1068]: could we add comments on what is actually happening here? And potentially also `expect` checks of what actually happened?

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx (hunk 4 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx]: done

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx (hunk 5 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx]: done

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx (hunk 7 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx:1068]: added comments and some expect checks

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx (hunk 6 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/tests/DomainExplorerManageColumns.test.tsx:584]: this click would trigger function a call which will in turn triggers query. have added some expect checks to it

**Review by jstastny-cz — APPROVED** [review]

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx [hunk 8]
```diff
@@ -300,40 +202,136 @@ const DomainExplorerManageColumns: React.FC<IOwnProps> = ({
...
+                allSelections.push(item.name + '/' + title + '/' + group.name);
+                return (
+                  <DataListContent
+                    aria-label={
+                      'Primary Content Details-' +
+                      group.name +
+                      '/' +
+                      title +
+                      '/' +
+                      item.name
```
Comment (cristianonicolai) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: can we have a variable to combine this. It seems to be recreated a few times and sometimes _group.name_ comes first, then _item.name_. would make it consistent.

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx [hunk 9]
```diff
@@ -343,31 +341,71 @@ const DomainExplorerManageColumns: React.FC<IOwnProps> = ({
...
                     }
                   })
-                  .map((item, _index) => (
-                    <SelectOption key={_index} value={item.name + group.name}>
-                      {item.name}
-                    </SelectOption>
-                  ))}
-            </SelectGroup>
+                  .map((item, _index) => {
+                    allSelections.push(item.name + '/' + group.name);
```
Comment (cristianonicolai) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: same here, create a variable to reuse this key, and I guess it should be `group.name + '/' + item.name`

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx [hunk 10]
```diff
@@ -379,61 +417,240 @@ const DomainExplorerManageColumns: React.FC<IOwnProps> = ({
...
+          const rest = filterColumnSelection(arr, objValue);
+          finalObj.push(rest);
+        });
+        finalObj.push(metaData);
+        setParameters(finalObj);
+      }
+    };
+
+    const items = [
+      <DropdownItem key={0} onClick={() => handleSelectClick('none')}>
```
Comment (cristianonicolai) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: Isnt better to split there into two separeted functions? `handleSelectNone` and `handleSelectAll` ?

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx [hunk 11]
```diff
@@ -330,7 +336,15 @@ const DomainExplorerTable = ({
                   No columns selected
                 </Title>
                 <EmptyStateBody>
-                  Select columns from the dropdown to see content
+                  <Button
+                    variant="link"
+                    id="retry-columns-button"
+                    onClick={handleRetry}
```
Comment (cristianonicolai) [inline ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx:339]: could you just have a function here let's say _onManageColumnsClick_ that is received as parameter that you don't need to do a set true?

**Review by nmirasch — APPROVED** [review]

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx (hunk 8 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: done

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx (hunk 9 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: done

**Comment by srambach** [comment]
Although Axe is happy now, when voiceover reads you can see some problems with what is described. Here are two fixes that should improve it.
image

**Comment by srambach** [comment]
BTW, "Primary Content Details-" appears a few different places and seems to be a placeholder that should be removed, as it just reads those words many times.

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx (hunk 10 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerManageColumns/DomainExplorerManageColumns.tsx]: done

File: ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx (hunk 11 above)
Comment (Sara4994) [reply ui-packages/packages/common/src/components/Molecules/DomainExplorerTable/DomainExplorerTable.tsx:339]: done

**Comment by Sara4994** [comment]
Jenkins please retest this

**Review by cristianonicolai — APPROVED** [review]

**Review by srambach — APPROVED** [review]
Please create a jira for going over the a11y again since the aria labels seem a bit off, but the rest looks good. Thank you for giving some time to maintaining good a11y!

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Pull request merged**